lib/improver/weighted_blend.py

```diff
@@ -91,8 +298,27 @@ def process(self, cube, weights=None):
             warnings.warn(msg)
             cube = iris.util.new_axis(cube, self.coord)
             collapse_dim = cube.coord_dims(self.coord)
-        # Supply weights as an array of weights whose shape matches the cube.
-        weights_array = None
+
+        # Check to see if the data is percentile data
+        perc_coord = None
+        perc_dim = None
+        perc_found = 0
+        for coord in cube.coords():
+            if coord.name().find('percentile') >= 0:
+                perc_found += 1
+                perc_coord = coord
```

Comment:
I think Caroline's way is fine. We could pass in the percentile coordinate, but I think this way works fine.